addons/web/static/src/js/components/datepicker.js

```diff
@@ -224,6 +225,7 @@ odoo.define('web.DatePickerOwl', function (require) {
         constructor() {
             super(...arguments);
 
+            this.pickerOptions.format = time.getLangDatetimeFormat();
```

Comment:
This would have already been done by the call to `super()` on the previous line, and would erase any format from the `props` argument.

Reply:
My bad: the spec was indeed to be able to override any of the keys given to the bootstrap object. It has been fixed and the picker object should now be updated in real time when props are changed.